### ISSUE TITLE
Nan Values on validation

### ISSUE THREAD

**Issue by yassineza77** [ultralytics/ultralytics#8422]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Val

### Bug

i trained yolov8n version:8.1.15 on my custom dataset that i preprocessed to the size of 416*416 and normalized using scaling [0,1]. I'm now facing an issue with validation process; on my dataset that is carefully curated i got for 2 epochs using the command:"yolo task=detect mode=train data=/home/Data.yaml model=yolov8n.pt epochs=2 batch=32 imgsz=640" the results were: "
wandb:          train/box_loss 1.66723
wandb:          train/cls_loss 2.90567
wandb:          train/dfl_loss 1.93911
wandb:            val/box_loss nan
wandb:            val/cls_loss nan
wandb:            val/dfl_loss nan" 
the issue is not with wandb cuz nothing changed after disabling it and not with number of epochs since i run the training for 100 epochs with no difference on the 'Nan' values. my dataset is completely right with no corrupt files and to test this, i run the training on COCO8 dataset, the results were: 
"wandb: model/speed_PyTorch(ms) 43.344
wandb:          train/box_loss nan
wandb:          train/cls_loss nan
wandb:          train/dfl_loss nan
wandb:            val/box_loss nan
wandb:            val/cls_loss nan
wandb:            val/dfl_loss nan
" even worse, i also updated ultralytics to yolo v8.1.18, in vain.
my configuration is as follows: Python-3.10.12 torch-2.2.0+cu11.8 CUDA:0 (NVIDIA GeForce GTX 1660 SUPER, 5929MiB) with OpenCV-DNN module with CUDA v11.8 backend support
P.S: i tried many batch sizes, i also tried reducing images in all dataset directories; train, val & test. no results.

### Environment

Ultralytics YOLOv8.1.18 🚀 Python-3.10.12 torch-2.2.0+cu118 CUDA:0 (NVIDIA GeForce GTX 1660 SUPER, 5929MiB)

### Minimal Reproducible Example
 for my custom dataset : yolo task=detect mode=train data=/home/Data.yaml model=yolov8n.pt epochs=2 batch=32 imgsz=640
for coco8 dataset: yolo task=detect mode=train model=yolov8n.pt epochs=2 batch=32 imgsz=640

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by glenn-jocher** [comment]
@yassineza77 hey there! Thanks for the update and for providing the reproducible code using terminal commands. It's really helpful! 🙌

Just to make sure we're on the same page, could you confirm that you've pulled the latest updates with `git pull` or updated the package using `pip install -U ultralytics`? This step is crucial to ensure that the issue isn't already addressed in the most recent codebase.

If you've already done that and the issue persists, it would be great if you could open a new issue using the Bug Report template. This way, we can track and investigate the problem more efficiently.

Looking forward to your new issue, and we'll do our best to sort this out quickly! 👍

**Comment by chandra-ps612** [comment]
Hello @glenn-jocher 
While experimenting with custom dataset using ```yolov8n.yaml``` on ```Ubuntu 22.10```, I encountered the similar issue. Here are the details below-
ultralytics == 8.1.29
torch == 2.2.1 + Cu12.1
torchaudio == 2.2.1 + Cu12.1
torchvision == 0.17.1 + Cu12.1
torchmetrics == 1.3.2
NVIDIA GeForce GTX 1650 (4 Gb)
```default.yaml``` changes-
amp : False
pretrained : False
optimizer : SGD
batch : 24
imgsz : 416
It is quite surprising after every epoch, NaN values are being generated as validation loss, albeit training loss has no issue including mAP@0.5, mAP@0.5:0.95, P, and R.
I also tried PyTorch package for CUDA 11.8 but couldn't manage to fix this issue.
Any help and suggestions is valuable!
Thanks

**Comment by glenn-jocher** [comment]
@fosteman thank you for sharing that helpful tip! Yes, insufficient disk space can cause NaN values during training as model weights fail to save properly. I recommend everyone check their available disk space before training and use `df -h` on Linux/Mac or check disk space on Windows to ensure adequate storage. If others encounter this issue, they can resume training from the last valid checkpoint using:

```python
model.train(resume=True)
```

For more details on handling training interruptions and checkpoints, see the Training guide at https://docs.ultralytics.com/modes/train/#resuming-training
